### PR TITLE
AEA-1467 Run locally and Migrate to Platform Hosting

### PR DESCRIPTION
**Pull request by booshibooshi**
Includes changes from #46

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,9 +31,5 @@
     "typescript": "4.3.5",
     "webpack": "5.44.0",
     "webpack-cli": "4.7.2"
-  },
-  "engines": {
-    "node": "12.x",
-    "npm": "7.x"
   }
```
Comment (booshibooshi) [inline package.json:34]: reinstate

File: client/index.js [hunk 2]
```diff
@@ -786,21 +783,29 @@ function getCanceller(cancelResponse, success) {
...
 
+function getPractitionerName(practitioner) {
+  const practitionerNameEntry = practitioner.name[0]
+  const practitionerNameText = practitioner.name[0].text
+  const practitionerName = practitionerNameText
+    ? practitionerNameText
+    : `${practitionerNameEntry.prefix[0]} ${practitionerNameEntry.given[0]} ${practitionerNameEntry.family}`
+  return practitionerName
+}
+
```
Comment (callumtipper-IW) [inline client/index.js:801]: Use existing name formatter?

File: ecs-proxies-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
+docker_service_cpu: 512
+docker_service_memory: 1024
+docker_service_autoscaling_prod:
+    service_metric: ALBRequestCountPerTarget
+    target_value: 1800
```
Comment (callumtipper-IW) [inline ecs-proxies-deploy.yml:5]: remove autoscaling?

File: ecs-proxies-deploy.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            valueFrom: "/{{ account }}/api-deployment/eps-api-tool/{{ APIGEE_ENVIRONMENT }}/test-client/rss-jwt-key-id"
+          - name: RSS_JWT_SUBJECT
+            valueFrom: "/{{ account }}/api-deployment/eps-api-tool/{{ APIGEE_ENVIRONMENT }}/test-client/rss-jwt-subject"
+          - name: RSS_JWT_PRIVATE_KEY
+            valueFrom: "arn:aws:secretsmanager:eu-west-2:{{ PTL_ACCOUNT_ID }}:secret:{{ account }}/eps-api-tool/{{ APIGEE_ENVIRONMENT }}/test-client/rss-jwt-private-key"
+          - name: DEMO_APP_PRIVATE_KEY
+            valueFrom: "arn:aws:secretsmanager:eu-west-2:{{ PTL_ACCOUNT_ID }}:secret:{{ account }}/eps-api-tool/{{ APIGEE_ENVIRONMENT }}/test-client/local-jwt-private-key"
+          # currently broken
+          # - name: DEMO_APP_CLIENT_SECRET
+          #   valueFrom: "arn:aws:secretsmanager:eu-west-2:{{ PTL_ACCOUNT_ID }}:secret:{{ account }}/eps-api-tool/{{ APIGEE_ENVIRONMENT }}/test-client/client-secret"
```
Comment (callumtipper-IW) [inline ecs-proxies-deploy.yml:52]: Check this

File: requirements.txt [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
 flask-compress
-flask_sqlalchemy
+flask-session
 flask_migrate
 httpx
-uwsgi
 cryptography
 jwt
-psycopg2
+psycopg2-binary
```
Comment (callumtipper-IW) [inline requirements.txt:8]: postgres - delete this

File: server/api.py [hunk 6]
```diff
@@ -58,23 +45,29 @@ def make_eps_api_request(path, access_token, body):
 
 
 def make_sign_api_signature_upload_request(auth_method, access_token, digest, algorithm):
+    if config.ENVIRONMENT.endswith("-sandbox"):
+        return {
+          "redirectUri": f'{config.BASE_URL}prescribe/send'
+        }
```
Comment (callumtipper-IW) [inline server/api.py:51]: remove all sandbox stuff?

File: server/api.py [hunk 7]
```diff
@@ -92,15 +85,21 @@ def make_sign_api_signature_upload_request(auth_method, access_token, digest, al
...
     print("Sending Signing Service signature upload request...")
 
-    return httpx.post(
+    response = httpx.post(
         f"{signing_base_url}/signaturerequest",
         headers={"Content-Type": "text/plain", "Authorization": f"Bearer {access_token}"},
         data=jwt_request,
         verify=False,
     ).json()
+    return response
```
Comment (callumtipper-IW) [inline server/api.py:94]: ?

File: server/api.py [hunk 8]
```diff
@@ -92,15 +85,21 @@ def make_sign_api_signature_upload_request(auth_method, access_token, digest, al
...
 
 
 def make_sign_api_signature_download_request(auth_method, access_token, token):
+    if config.ENVIRONMENT.endswith("-sandbox"):
+        return {
+          "signature": "",
+          "certificate": ""
+        }
```
Comment (callumtipper-IW) [inline server/api.py:102]: remove sandbox ?

File: server/app.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+from cryptography.fernet import Fernet
+from flask import Flask
+from flask_migrate import Migrate
+from flask_compress import Compress
+import config
+from flask_session import Session
+import redis
+
+fernet = Fernet(config.SESSION_TOKEN_ENCRYPTION_KEY)
+migrate = Migrate()
```
Comment (callumtipper-IW) [inline server/app.py:11]: postgres - remove?

File: store/healthcheck.go [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    msg := "OK"
+    _, err := w.Write([]byte(msg))
+    if err != nil {
+        fmt.Printf("Failed to write response, err: %s", err)
+    }
+}
+
+func main() {
+    StartWebServer()
+}
```
Comment (callumtipper-IW) [inline store/healthcheck.go:37]: 1. using Go? can we use TypeScript?
2. Is this healthcheck really checking health?

**Review by callumtipper-IW — COMMENTED** [review]
see further comments

File: server/api.py (hunk 6 above)
Comment (booshibooshi) [reply server/api.py:51]: Do we have an alternative for running locally?

File: server/api.py (hunk 8 above)
Comment (booshibooshi) [reply server/api.py:102]: Do we have an alternative for running locally?

File: store/healthcheck.go (hunk 10 above)
Comment (booshibooshi) [reply store/healthcheck.go:37]: as discussed last Friday, using go as it was the smallest thing to augment redis docker image with a healthcheck to keep the ECS task eps-api-tool alive, once this pr is merged we can all contribute in parallel and change this to whatever we want in future but before this is merged we have to take it in turns to develop

**Review by mfjarvis — APPROVED** [review]

**Pull request merged**